src/Security/Tls/AppleTlsProvider.cs

```diff
@@ -21,23 +22,29 @@ namespace XamCore.Security.Tls
 {
 	class AppleTlsProvider : MonoTlsProvider
 	{
-		static readonly Guid id = new Guid ("981af8af-a3a3-419a-9f01-a518e3a17c1c");
-
 		public override string Name {
 			get { return "apple-tls"; }
 		}
 
 		public override Guid ID {
-			get { return id; }
+			get { return MNS.MonoTlsProviderFactory.AppleTlsId; }
 		}
 
 		public override IMonoSslStream CreateSslStream (
 			Stream innerStream, bool leaveInnerStreamOpen,
 			MonoTlsSettings settings = null)
 		{
-			return new AppleTlsStream (innerStream, leaveInnerStreamOpen, settings, this);
+			return SslStream.CreateMonoSslStream (innerStream, leaveInnerStreamOpen, this, settings);
```

Comment:
@baulig ^ can you confirm this is right ? thanks!

Reply:
Yes, that's right.